driver-core/src/main/java/com/datastax/driver/core/querybuilder/BuiltStatement.java

```diff
@@ -231,7 +231,7 @@ void maybeAddRoutingKey(String name, Object value) {
             return;
 
         for (int i = 0; i < partitionKey.size(); i++) {
-            if (name.equals(partitionKey.get(i).getName())) {
+            if (name.equalsIgnoreCase(partitionKey.get(i).getName())) {
```

Comment:
Great catch!  I have one worry here which has to do with quoted identifiers.  For example you can have a table with the following schema:

```
CREATE TABLE simple.blah (
    "hI" int PRIMARY KEY,
    "Hi" int,
    hi int
)
```

By using equalsIgnoreCase, `"Hi"` would match against `"hI"` and be considered part of the routing key values, which we don't want.  I think to work around this you could use:

```java
Metadata.handleId(name).equals(partitionKey.get(i).getName())
```